lib/commands/connect/auth_connection.js

```diff
@@ -27,27 +29,24 @@ module.exports = {
     ],
     needsApp: true,
     needsAuth: true,
-    run: cli.command(function (context, heroku) {
-      return api.withUserConnections(context.auth.password, context.app, context.flags, true, heroku)
-        .then(function(connections) {
-          if (connections.length > 0) {
-            callbackServer();
+    run: cli.command(co.wrap(function* (context, heroku) {
+      let connections = yield api.withConnection(context, heroku);
+      callbackServer();
+
+      let url = '/api/v3/connections/' + connection.id + '/authorize_url';
```

Comment:
I believe its safe to use template strings as well (if you want).
``` javascript
let url = `/api/v3/connections/${connection.id}/authorize_url`;
```

Assuming [this feature list](https://nodejs.org/en/docs/es6/#which-features-ship-with-node-js-by-default-no-runtime-flag-required) is a good reference.